modules/react/popup/lib/hooks/useReturnFocus.tsx

```diff
@@ -14,13 +27,25 @@ import {usePopupModel} from './usePopupModel';
 export const useReturnFocus = createElemPropsHook(usePopupModel)(model => {
   const visible = model.state.visibility !== 'hidden';
 
-  React.useLayoutEffect(() => {
+  React.useEffect(() => {
```

Comment:
`useEffect` must be used for `document.activeElement` to be the correct element. `useLayoutEffect` happens before the browser shifts focus.

The downside is there may be a frame rendered by the browser where the target element doesn't have focus. The upside is we can add extra heuristics to determine if the user clicked on a focusable element (we should not change focus in this case).